### PR TITLE
[EXPERIMENTAL] Next.js 13 `app` directory example

### PR DESCRIPTION
**Pull request by thorwebdev**

### CODE REVIEW

File: examples/nextjs/app/nextjs-13/ssr/page.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      // const newSetCookies = filterCookies(
+      //   ensureArray(context.res.getHeader('set-cookie')?.toString() ?? []),
+      //   name
+      // );
+      // const newSessionStr = serializeCookie(name, value, {
+      //   ...options,
+      //   // Allow supabase-js on the client to read the cookie as well
+      //   httpOnly: false
+      // });
+      // context.res.setHeader('set-cookie', [...newSetCookies, newSessionStr]);
```
[the remark this answers is not in the document]
Comment (thorwebdev) [reply examples/nextjs/app/nextjs-13/ssr/page.tsx:52]: These are read-only functions: https://beta.nextjs.org/docs/upgrade-guide#accessing-request-object

But we're able to modify the res via middleware, see discussion here: https://github.com/supabase/auth-helpers/issues/341#issuecomment-1292658432

File: examples/nextjs/app/nextjs-13/client-component/page.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Next.js 13 Client Component
+// NOTE: This is experimental and used as research to understand how to support Next.js 13 in the future.
+// https://beta.nextjs.org/docs/data-fetching/fetching#example-fetch-and-use-in-client-components
+
+import { use } from 'react';
```
Comment (imbhargav5) [inline examples/nextjs/app/nextjs-13/client-component/page.tsx:5]: 🤔 Will this be a client component without `use client`?

**Comment by cdedreuille** [comment]
It would be great to keep the old example with the `pages` folder as it will probably stay for a while until the full transition to `app` is done.

**Comment by thorwebdev** [comment]
With version `v0.5.2` we've added a `createServerComponentSupabaseClient` which can be used server components to get an authenticated Supabase client. [This is the PR](https://github.com/supabase/auth-helpers/pull/380) which includes an [example](https://github.com/supabase/auth-helpers/blob/main/examples/nextjs-server-components/app/required-session/page.tsx). Note that this needs to be used in combination with [middleware](https://github.com/supabase/auth-helpers/blob/main/examples/nextjs-server-components/middleware.tsx) as Next.js does not yet allow setting headers in server components.

**Pull request closed without merging**